src/Tickets/Order_Modifiers/Modifiers/Coupon.php

```diff
@@ -0,0 +1,150 @@
+<?php
+/**
+ * Concrete Strategy for Coupon Modifiers.
+ *
+ * Handles the specific logic for Coupon modifiers, including inserting, updating,
+ * rendering, and validating coupon data.
+ *
+ * @since TBD
+ *
+ * @package TEC\Tickets\Order_Modifiers\Modifiers;
+ */
+
+namespace TEC\Tickets\Order_Modifiers\Modifiers;
+
+use TEC\Tickets\Order_Modifiers\Repositories\Order_Modifiers as Order_Modifiers_Repository;
+use TEC\Tickets\Order_Modifiers\Models\Order_Modifier;
+
+/**
+ * Concrete Strategy for Coupon Modifiers.
+ *
+ * @since TBD
+ */
+class Coupon implements Modifier_Strategy_Interface {
+
+	/**
+	 * The modifier type for coupons.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	protected string $modifier_type = 'coupon';
+
+	/**
+	 * Gets the modifier type for coupons.
+	 *
+	 * @since TBD
+	 *
+	 * @return string The modifier type ('coupon').
+	 */
+	public function get_modifier_type(): string {
```

Comment:
This would be better as part of an `abstract` class.

Reply:
In my other branch I moved the majority of logic from the Coupon class into a new abstract class.  https://github.com/the-events-calendar/event-tickets/blob/feature/ET-2203_coupon_edit_screen/src/Tickets/Order_Modifiers/Modifiers/Modifier_Abstract.php#L75